### PR TITLE
EKS - Provisioning EKS Control Plane

### PR DESCRIPTION
**Pull request by carlosgit2016**
- Created required IAM Roles for Master Nodes
- Provisioning EKS Cluster without The Worker Nodes

### CODE REVIEW

**Comment by carlosgit2016** [comment]
With some workaround, the subnets for each AZ 
```terraform
  # aws_subnet.eks_subnets["us-east-1a"] will be created
  + resource "aws_subnet" "eks_subnets" {
      + arn                                            = (known after apply)
      + assign_ipv6_address_on_creation                = false
      + availability_zone                              = "us-east-1a"
      + availability_zone_id                           = (known after apply)
      + cidr_block                                     = "10.0.1.0/24"
      + enable_dns64                                   = false
      + enable_resource_name_dns_a_record_on_launch    = false
      + enable_resource_name_dns_aaaa_record_on_launch = false
      + id                                             = (known after apply)
      + ipv6_cidr_block_association_id                 = (known after apply)
      + ipv6_native                                    = false
      + map_public_ip_on_launch                        = false
      + owner_id                                       = (known after apply)
      + private_dns_hostname_type_on_launch            = (known after apply)
      + tags_all                                       = {
          + "owners"  = "minders"
          + "project" = "master-the-legacy"
        }
      + vpc_id                                         = "vpc-0695e38849351ddc9"
    }

  # aws_subnet.eks_subnets["us-east-1b"] will be created
  + resource "aws_subnet" "eks_subnets" {
      + arn                                            = (known after apply)
      + assign_ipv6_address_on_creation                = false
      + availability_zone                              = "us-east-1b"
      + availability_zone_id                           = (known after apply)
      + cidr_block                                     = "10.0.2.0/24"
      + enable_dns64                                   = false
      + enable_resource_name_dns_a_record_on_launch    = false
      + enable_resource_name_dns_aaaa_record_on_launch = false
      + id                                             = (known after apply)
      + ipv6_cidr_block_association_id                 = (known after apply)
      + ipv6_native                                    = false
      + map_public_ip_on_launch                        = false
      + owner_id                                       = (known after apply)
      + private_dns_hostname_type_on_launch            = (known after apply)
      + tags_all                                       = {
          + "owners"  = "minders"
          + "project" = "master-the-legacy"
        }
      + vpc_id                                         = "vpc-0695e38849351ddc9"
    }
```

File: infrastructure/scaffold/networking.tf [hunk 1]
```diff
@@ -9,8 +15,8 @@ resource "aws_vpc" "legacy-vpc" {
 # EKS subnets
 # Need to be at least two subnets in two different AZs
 resource "aws_subnet" "eks_subnets" {
-  count             = 2
+  for_each = local.eks_availability_zones
   vpc_id            = aws_vpc.legacy-vpc.id
-  cidr_block        = "10.0.${count.index + 1}.0/24"
-  availability_zone = "us-east-1a"
+  cidr_block        = "10.0.${index(tolist(local.eks_availability_zones), each.value) + 1}.0/24"
+  availability_zone = each.value # Ensure multi zones
```
Comment (benhur-araujo) [inline infrastructure/scaffold/networking.tf:21]: Nice adjustment here

File: infrastructure/scaffold/eks.tf [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    #aws_iam_role_policy_attachment.example-AmazonEKSVPCResourceController
+  ]
+
+  # https://kubernetes.io/releases/
+  # Supported versions by EKS https://docs.aws.amazon.com/eks/latest/userguide/kubernetes-versions.html
+  version = "1.27" # no upgrades will occur except those automatically triggered by EKS
+
+  tags = {
+    product = "master-the-legacy"
+    ctf     = "WW91IGdvdCBpdCEgbmljZS4="
```
Comment (benhur-araujo) [inline infrastructure/scaffold/eks.tf:48]: what is this?

**Review by benhur-araujo — APPROVED** [review]

File: infrastructure/scaffold/eks.tf (hunk 2 above)
Comment (carlosgit2016) [reply infrastructure/scaffold/eks.tf:48]: A challenge to you figure out

**Pull request merged**